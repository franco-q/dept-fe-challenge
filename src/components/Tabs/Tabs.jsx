import cn from 'classnames'
import { useState } from 'react'
import css from './Tabs.module.scss'

const Tabs = ({ links, onClickLink }) => {
  const [active, setActive] = useState(links[0])

  const onClick = link => e => {
    e.preventDefault()
    setActive(link)
    onClickLink(link)
  }

  return (
    <nav className={css.root}>
      {links.map((link, i) => (
        <a
          href="#"
          className={cn(css.tab, active === link && css.active)}
          onClick={onClick(link)}
          key={i}
        >
          {link}
        </a>
      ))}
    </nav>
  )
}

export default Tabs
