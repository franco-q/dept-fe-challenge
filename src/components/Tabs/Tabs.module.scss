@import '@/scss/styles.scss';

.root {
  background: linear-gradient(180deg, #121212 64.11%, #1e1e1e 100%);
  box-shadow: 0px 8px 10px rgba(0, 0, 0, 0.14), 0px 3px 14px rgba(0, 0, 0, 0.12),
    0px 5px 5px rgba(0, 0, 0, 0.2);
  height: 48px;
  display: flex;
}

.tab {
  font-family: 'D-DIN';
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  text-align: center;
  position: relative;
  flex: 1 1 auto;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  letter-spacing: 0.05em;
  &,
  &:active,
  &:hover {
    color: rgba(255, 255, 255, 0.87);
  }
  &.active::after {
    content: '';
    display: block;
    position: absolute;
    top: 100%;
    height: 2px;
    width: 100%;
    background: rgba(255, 255, 255, 0.87);
  }
}

@include media-breakpoint-up(tablet) {
  .tab {
    max-width: 180px;
  }
}
