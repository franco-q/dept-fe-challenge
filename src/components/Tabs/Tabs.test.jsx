import { render, fireEvent } from '@testing-library/react'
import Tabs from './Tabs'

jest.mock('react-router-dom', () => ({
  Link: ({ children, to }) => {
    return <a href={to}>{children}</a>
  }
}))

describe('Tabs', () => {
  test('should render', () => {
    const props = { links: ['All', 'Favourites'] }

    const { asFragment } = render(<Tabs {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should execute cb when a link is clicked', () => {
    const onClickLink = jest.fn()
    const props = { links: ['All', 'Favourites'] }
    const { container, getByText } = render(
      <Tabs {...props} onClickLink={onClickLink} />
    )

    fireEvent.click(getByText(props.links[0]))

    expect(onClickLink).toHaveBeenCalledWith(props.links[0])
  })
})
