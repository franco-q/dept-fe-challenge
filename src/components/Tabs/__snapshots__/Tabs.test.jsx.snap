// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render 1`] = `
<DocumentFragment>
  <nav>
    <a
      class=""
      href="#"
    >
      All
    </a>
    <a
      class=""
      href="#"
    >
      Favourites
    </a>
  </nav>
</DocumentFragment>
`;
